packages/back-end/src/integrations/BigQuery.ts

```diff
@@ -151,10 +151,19 @@ export default class BigQuery extends SqlIntegration {
       database
     );
   }
+
+  async listDatasets(): Promise<(string | undefined)[]> {
```

Comment:
Shouldn't we always have `string` here? Feels like we could have errors in `getInformationSchema` if for some reason there was an undefined result among actual datasets.

Maybe filter would be better here?

Reply:
@lukesonnet Good suggestion - I cleaned this up a bit.

Ultimately, it feels like there is a valid (though, weird) case where a project might not have any datasets. In that case, rather than waiting until we see if we get any results (which we know we're not going to) - I'm throwing an error with a more contextual message (No datasets found).

I've also cleaned up the code so we only ever return an array of strings. The `bg.Dataset` type has a pretty funky type `ServiceObject<any>.id?: string | undefined` that has been giving me trouble, especially with the `filter` method - in order to get it to work I had to assert the id was a string which smelled a bit to me.

I went old school with a forLoop.

Let me know what you think.

Reply:
I feel like a map like you had it before, followed by a filter would work, but this looks good as well!